### PR TITLE
Config - add support for environment variables 

### PR DESCRIPTION
**Pull request by na-ji**
Hello,

I added the possibility to configure Golbat from environment variables. To achieve this goal, I used the library [koanf](https://github.com/knadh/koanf). Also, I decided to use Unmarshal, this way the implementation of the new config didn't require updating all our usages. 

The main challenge when implementing this PR was reading slices on environment variables. This required to write a custom envParser, since the default one was overwriting slice members with each others.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -3,89 +3,89 @@ package config
...
+	Db       string `koanf:"db"`
+	MaxPool  int    `koanf:"max_pool"`
 }
 
 type tuning struct {
-	ExtendedTimeout bool `toml:"extended_timeout"`
-	ProcessWilds    bool `toml:"process_wild_pokemon"`
-	ProcessNearby   bool `toml:"process_nearby_pokemon"`
+	ExtendedTimeout bool `koanf:"extended_timeout"`
+	ProcessWilds    bool `koanf:"process_wild_pokemon"`
```
Comment (jfberry) [inline config/config.go:87]: I think this is removed now

File: config/config.go [hunk 2]
```diff
@@ -3,89 +3,89 @@ package config
...
+	ProcessNearby   bool `koanf:"process_nearby_pokemon"`
 }
 
 type scanRule struct {
```
Comment (jfberry) [inline config/config.go:91]: this doesn't appear to have been converted

File: docker-compose.yml.example [hunk 3]
```diff
@@ -4,17 +4,64 @@ services:
...
+      - GOLBAT.DATABASE.USER=
+      - GOLBAT.DATABASE.PASSWORD=
+      - GOLBAT.DATABASE.ADDRESS=host.docker.internal:3306
+      - GOLBAT.DATABASE.DB=
+
+      - GOLBAT.PVP.ENABLED=true
+      - GOLBAT.PVP.INCLUDE_HUNDOS_UNDER_CAP=false
+      - GOLBAT.PVP.LEVEL_CAPS=50,51
+
+      - GOLBAT.PVP.LEAGUES.0.NAME=little
```
Comment (jfberry) [inline docker-compose.yml.example:47]: I wonder whether we should take the opportunity to tidy up pvp configuration as these are just trash, especially here in docker config. We could do as a seperate PR (I would be happy to do first and you merge up?) or we could do as part of this one

Comment (na-ji) [reply docker-compose.yml.example:47]: You mean setting default values or removing the config altogether?

File: config/reader.go [hunk 4]
```diff
@@ -1,54 +1,83 @@
...
+		Logging: logging{
+			SaveLogs: false,
+		},
+		Cleanup: cleanup{
+			StatsDays: 7,
+		},
+		Database: database{
+			MaxPool: 100,
+		},
+		Tuning: tuning{
```
[the remark this answers is not in the document]
Comment (na-ji) [reply config/reader.go:37]: Shit, it's a rebasing fail. I will fix it.

File: docker-compose.yml.example (hunk 3 above)
Comment (jfberry) [reply docker-compose.yml.example:47]: as per malte's issue discussion (or small variation) - getting rid of the config altogether. The leagues are the leagues, the only real option is whether the 500 lead has the evolution restriction (and the global settings for level caps etc, for which there would be sensible defaults)

**Comment by jfberry** [comment]
Does koanf have same behaviour for the scan rules - ie null if not present but true / false in the bool pointer?
If it does then it looks great

**Comment by na-ji** [comment]
> Does koanf have same behaviour for the scan rules - ie null if not present but true / false in the bool pointer? If it does then it looks great

Yes, I can confirm that the behavior is the same as on main, by default the slice and the boolean pointers are nil. I've tried with the config you provided #90.

**Comment by jfberry** [comment]
Let’s go to market :-)

**Comment by na-ji** [comment]
> Let’s go to market :-)

Alright, thank you 😄

**Pull request merged**